src/math/sqrt_int16.c

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+/* Include Files */
+#include <sof/math/sqrt.h>
+
+/* Defines Constant*/
+#define SQRT_WRAP_SCHAR_BITS 0xFF
+
+/* Function Definitions */
+/*
+ *  Square root
+ *
+ *  Y = SQRTLOOKUP_INT16(U) computes the square root of
+ *  U using lookup tables.
+ * +------------------+-----------------+--------+--------+
+ * | u		      | y (returntype)	|   u	 |  y	  |
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * | 16 | 12  |	 0    | 0  | 15 |   1	| 4.12	 | 1.15	  |
```

Comment:
Wrong Qformat comment. Based on my test with the code the input seems to be Q0.16 unsigned and output Q4.12 signed compatible.